## Papers on machine learning for proteins

### Background

We recently released a [review](https://arxiv.org/abs/1811.10775) of machine learning methods in protein engineering, but the field changes so fast and there are so many new papers that any static document will inevitably be missing important work. We hope that this will be a useful resource for people interested in the field. 

To the best of our knowledge, this is the first public, collaborative list of machine learning papers on protein applications. We try to classify papers based on a combination of their applications and model type. If you have suggestions for other papers or categories, please make a pull request or issue!

### Format

Within each category, papers are listed in reverse chronological order (newest first). Where possible, a link should be provided. 

### Categories

[Reviews](#reviews)  
[Machine-learning guided directed evolution](#machine-learning-guided-directed-evolution)  
[Representation learning](#representation-learning)  
[Unsupervised variant prediction](#unsupervised-variant-prediction)  
[Generative models](#generative-models)  
[Predicting stability](#predicting-stability)  
[Predicting structure from sequence](#predicting-structure-from-sequence)  
[Predicting sequence from structure](#predicting-sequence-from-structure)  
[Classification and annotation](#classification-and-annotation)  
[Predicint nteractions with other molecules](#predicting-interactions-with-other-molecules)  
[Other supervised learning](#other-supervised-learning)

### Reviews

**Machine learning-guided directed evolution for protein engineering.**  
Kevin K. Yang, Zachary Wu, Frances H. Arnold.   
*Preprint, April 2019.*   
[[arxiv](https://arxiv.org/abs/1811.10775)]

### Machine-learning guided directed evolution

**Batched Stochastic Bayesian Optimization via Combinatorial Constraints Design**.   
Kevin K. Yang, Yuxin Chen, Alycia Lee, Yisong Yue.   
*International Conference on Artificial Intelligence and Statistics (AISTATS), April 2019.*  
[[arxiv](https://arxiv.org/abs/1904.08102)]

**Machine learning-assisted directed protein evolution with combinatorial libraries.**  
Zachary Wu, S. B. Jennifer Kan, Russell D. Lewis, Bruce J. Wittmann, Frances H. Arnold.  
*PNAS, April 2019.*  
[[10.1073/pnas.1901979116](https://doi.org/10.1073/pnas.1901979116)]

**Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics.**  
Claire N. Bedbrook, Kevin K. Yang, J. Elliott Robinson, Viviana Gradinaru, Frances H .Arnold.  
*Preprint, March 2019.*  
[[10.1101/565606](https://doi.org/10.1101/565606)] [[bioRxiv](https://www.biorxiv.org/content/10.1101/565606v1)]

**Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization.**  
Claire N. Bedbrook, Kevin K. Yang, Austin J. Rice, Viviana Gradinaru, Frances H. Arnold.  
*PLOS Computational Biology, October 2017.*  
[[10.1371/journal.pcbi.1005786](https://doi.org/10.1371/journal.pcbi.1005786)]

**Exploring sequence-function space of a poplar glutathione transferase using designed information-rich gene variants.**  
Yaman Musdal, Sridhar Govindarajan, Bengt Mannervik.  
*Protein Engineering, Design, and Selection, August 2017.*  
[[10.1093%2Fprotein%2Fgzx045](https://dx.doi.org/10.1093%2Fprotein%2Fgzx045)]

**Navigating the protein fitness landscape with Gaussian processes.**  
Philip A. Romero, Andreas Krause, Frances H. Arnold.
*PNAS, January 2013.*  
[[10.1073/pnas.1215251110](https://doi.org/10.1073/pnas.1215251110)]

**Engineering proteinase K using machine learning and synthetic genes.**  
Jun Liao, Manfred K. Warmuth, Sridhar Govindarajan, Jon E. Ness, Rebecca P Wang, Claes Gustafsson, Jeremy Minshull.  
*BMC Biotechnology, March 2007.*  
[[10.1186/1472-6750-7-16](https://doi.org/10.1186/1472-6750-7-16)]

**Improving catalytic function by ProSAR-driven enzyme evolution.**  
Richard J. Fox, S. Christopher Davis, Emily C. Mundorff, Lisa M. Newman, Vesna Gavrilovic, Steven K. Ma, Loleta M. Chung, Charlene Ching, Sarena Tam, Sheela Muley, John Grate, John Gruber, John C. Whitman, Roger A. Sheldon, Gjalt W. Huisman.  
*Nature Biotechnology, February 2007.*  
[[Nature Biotechnology](https://www.nature.com/articles/nbt1286)]

### Representation learning

**Leveraging implicit knowledge in neural networks for functional dissection and engineering of proteins.**  
Julius Upmeier zu Belzen, Thore Bürgel, Stefan Holderbach, Felix Bubeck, Lukas Adam, Catharina Gandor, Marita Klein, Jan Mathony, Pauline Pfuderer, Lukas Platz, Moritz Przybilla, Max Schwendemann, Daniel Heid, Mareike Daniela Hoffmann, Michael Jendrusch, Carolin Schmelas, Max Waldhauer, Irina Lehmann, Dominik Niopek, Roland Eils.  
*Nature Machine Intelligence, May 2019.*  
[[Nature Machine Intelligence](https://www.nature.com/articles/s42256-019-0049-9)]

### Unsupervised variant prediction

### Generative models

### Predicting stability

**mGPfusion: predicting protein stability changes with Gaussian process kernel learning and data fusion.**
Emmi Jokinen, Markus Heinonen, Harri Lähdesmäki.  
*Bioinformatics, July 2018.*  
[[10.1093/bioinformatics/bty238](https://doi.org/10.1093/bioinformatics/bty238)]

**Structure Based Thermostability Prediction Models for Protein Single Point Mutations with Machine Learning Tools.**  
Lei Jia , Ramya Yarlagadda, Charles C. Reed.  
*PLOS One, September 2015.*  
[[10.1371/journal.pone.0138022](https://doi.org/10.1371/journal.pone.0138022)]

**mCSM: predicting the effects of mutations in proteins using graph-based signatures.**  
Douglas E. V. Pires, David B. Ascher, Tom L. Blundell.  
*Bioinformatics, February 2014.*  
[[10.1093/bioinformatics/btt691](https://doi.org/10.1093/bioinformatics/btt691)]

**PROTS-RF: A Robust Model for Predicting Mutation-Induced Protein Stability Changes.**  
Yunqi Li, Jianwen Fang.  
*PLOS One, October 2012.*  
[[10.1371/journal.pone.0047247](https://doi.org/10.1371/journal.pone.0047247)]

**Predicting changes in protein thermostability brought about by single- or multi-site mutations.**  
Jian Tian, Ningfeng Wu, Xiaoyu Chu, Yunliu Fan.  
*BMC Bioinformatics, July 2010.*  
[[10.1186/1471-2105-11-370](https://doi.org/10.1186/1471-2105-11-370)]

**Prediction of protein stability changes for single‐site mutations using support vector machines.**  
Jianlin Cheng, Arlo Randall, Pierre Baldi.  
*Proteins, December 2005*.  
[[10.1002/prot.20810](https://doi.org/10.1002/prot.20810)]

**Predicting protein stability changes from sequences using support vector machines.**  
Emidio Capriotti, Piero Fariselli, Remo Calabrese, Rita Casadio.  
*Bioinformatics, September 2005.*  
[[10.1093/bioinformatics/bti1109](https://doi.org/10.1093/bioinformatics/bti1109)]

**I-Mutant2.0: predicting stability changes upon mutation from the protein sequence or structure.**  
Emidio Capriotti, Piero Fariselli, Rita Casadio.  
*Nucleic Acids Research, July 2005.*  
[[10.1093/nar/gki375](https://doi.org/10.1093/nar/gki375)]

**Mismatch string kernels for discriminative protein classification.**  
Christina S. Leslie, Eleazar Eskin, Adiel Cohen, Jason Weston, William Stafford Noble.  
*Bioinformatics, March 2004.*  
[[10.1093/bioinformatics/btg431](https://doi.org/10.1093/bioinformatics/btg431)]


### Predicting structure from sequence

### Predicting sequence from structure

### Classification and annotation

**DeepGO: predicting protein functions from sequence and interactions using a deep ontology-aware classifier.**  
Maxat Kulmanov, Mohammed Asif Khan, Robert Hoehndorf.  
*Bioinformatics, February 2018.*  
[[Bioinformatics](https://doi.org/10.1093/bioinformatics/btx624)]

**The spectrum kernel: A string kernel for SVM protein classification.**  
Christina S Leslie, Eleazar Eskin, William Stafford Noble.  
*Pacific Symposium on Biocomputing, January 2002.*  
[[pdf](http://psb.stanford.edu/psb-online/proceedings/psb02/leslie.pdf)]

### Predicting interactions with other molecules

### Other supervised learning

**A statistical model for improved membrane protein expression using sequence-derived features.**  
Shyam M. Saladi, Nauman Javed, Axel Müller, William M. Clemons, Jr.  
*Journal of Biological Chemistry, March 2018.*  
[[10.1074/jbc.RA117.001052](https://doi.org/10.1074/jbc.RA117.001052)]

**Learning epistatic interactions from sequence-activity data to predict enantioselectivity.**  
Julian Zaugg, Yosephine Gumulya, Alpeshkumar K. Malde, Mikael Bodén.  
*Journal of Computer Aided Molecular Design, December 2017.*  
[[10.1007/s10822-017-0090-x](https://doi.org/10.1007/s10822-017-0090-x)]
